src/dcm230/test_dcm230.py

```diff
@@ -57,267 +57,105 @@ def test_range_validation() -> None:
-    client.read_input_registers.return_value = mock_result
-    value = meter.A
-    assert value == 230
-
-    """Test 2: Should pass."""
-    mock_result.registers = [0x4624, 0x1000]
-    client.read_input_registers.return_value = mock_result
-    value = meter.A
-    assert value == 10500
-
-
-def test_W() -> None:
-    """Test get w."""
-    client = MagicMock()
-    mock_result = MagicMock()
-    mock_result.isError.return_value = False
-    meter = Dcm230(1, client)
-
-    """Test 1: should pass"""
-    mock_result.registers = [0x4366, 0x0000]
-    client.read_input_registers.return_value = mock_result
-    value = meter.W
-    assert value == 230
-
-    """Test 2: Should pass."""
-    mock_result.registers = [0x4624, 0x1000]
-    client.read_input_registers.return_value = mock_result
-    value = meter.W
-    assert value == 10500
-
-
-def test_kwh() -> None:
-    """Test get kwh."""
-    client = MagicMock()
-    mock_result = MagicMock()
-    mock_result.isError.return_value = False
-    meter = Dcm230(1, client)
 
-    """Test 1: should pass"""
-    mock_result.registers = [0x4366, 0x0000]
-    client.read_input_registers.return_value = mock_result
-    value = meter.kwh
-    assert value == 230
+    """Test 1: Should pass."""
+    for name, spec in Dcm230._register_specs.items():  # type: ignore[attr-defined]
+        if spec.reg_type != 0x03:
+            continue
 
-    """Test 2: Should pass."""
-    mock_result.registers = [0x4624, 0x1000]
-    client.read_input_registers.return_value = mock_result
-    value = meter.kwh
-    assert value == 10500
+        value = getattr(meter, name)
+        assert value == 1
 
 
-def test_W_dmd() -> None:
-    """Test get w demnad."""
+def read_holding_registers() -> None:
+    """Test all holding registers."""
     client = MagicMock()
     mock_result = MagicMock()
     mock_result.isError.return_value = False
     meter = Dcm230(1, client)
 
-    """Test 1: should pass"""
-    mock_result.registers = [0x4366, 0x0000]
-    client.read_input_registers.return_value = mock_result
-    value = meter.W_dmd
-    assert value == 230
-
-    """Test 2: Should pass."""
-    mock_result.registers = [0x4624, 0x1000]
-    client.read_input_registers.return_value = mock_result
-    value = meter.W_dmd
-    assert value == 10500
-
-
-def test_W_dmd_peak() -> None:
-    """Test get w demand peak."""
-    client = MagicMock()
-    mock_result = MagicMock()
-    mock_result.isError.return_value = False
-    meter = Dcm230(1, client)
+    mock_result.registers = [0x3F80, 0x0000]
+    client.read_holding_registers.return_value = mock_result
 
-    """Test 1: should pass"""
-    mock_result.registers = [0x4366, 0x0000]
-    client.read_input_registers.return_value = mock_result
-    value = meter.W_dmd_peak
-    assert value == 230
+    """Test 1: Should pass."""
+    for name, spec in Dcm230._register_specs.items():  # type: ignore[attr-defined]
+        if spec.reg_type != 0x03:
```

Comment:
This seems wrong? Should be `if spec.reg_type != 0x04`?. Also you have the constants defined for register type, please use them.